pkg/utils/validation/kubernetesversion/version.go

```diff
@@ -20,8 +23,18 @@ var SupportedVersions = []string{
 	"1.34",
 }
 
+// EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV holds the name of the environment variable to prevent a crash
+// if the detected k8s version is not in the list of supported k8s versions.
+// This should only be used if you know exactly what you are doing and on your own risk.
+const EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV = "DO_NOT_CRASH_ON_UNSUPPORTED_KUBERNETES_VERSION"
+
 // CheckIfSupported checks if the provided version is part of the supported Kubernetes versions list.
 func CheckIfSupported(gitVersion string) error {
+	if os.Getenv(EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV) == "true" {
+		logf.Log.Info("Not checking for k8s version compatibility because flag is set to true", "flag", EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV)
```

Comment:
Cosmetic suggestion:
```suggestion
		logf.Log.Info("Skipping the check if the Kubernetes version is supported because flag is set to true", "flag", EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV)
```

Feel free to ignore if you prefer the initial version instead.

Reply:
Done via: [`c269f3b` (#13221)](https://github.com/gardener/gardener/pull/13221/commits/c269f3bb2070893a58301bddf272f289071250be)